File: NN-pythonV/predict.py
import numpy as np
from sigmoid import sigmoid

def predict(Theta1, Theta2, X):
	m = np.size(X, 0)
	num_labels = np.size((Theta2, 0))
	p = np.zeros((np.size((X, 1)), 1))

	h1 = sigmoid(np.concatenate((np.ones((m, 1)), X),axis=1).dot(Theta1.T))
	h2 = sigmoid(np.concatenate((np.ones((m, 1)), h1),axis=1).dot(Theta2.T))
	p = np.argmax(h2, axis=1).T
	#p = (np.amax(h2, axis=1)).T
	return p

File: NN-MatlabV/predict.m
function p = predict(Theta1, Theta2, X)
m = size(X, 1);
num_labels = size(Theta2, 1);
p = zeros(size(X, 1), 1);
h1 = sigmoid(double([ones(m, 1) X]) * double(Theta1'));
h2 = sigmoid(double([ones(m, 1) h1]) * double(Theta2'));
[dummy, p] = max(h2, [], 2);
end
